src/test/scala/io/citrine/lolo/transformers/FeatureRotatorTest.scala

```diff
@@ -0,0 +1,227 @@
+    val unrotatedInputs = FeatureRotator.applyRotation(rotatedInputs, featuresToRotate, U.t)
+    inputs.indices.foreach { i =>
+      assert(inputs(i)(0) == unrotatedInputs(i)(0), "Failed to leave categorical invariant.")
+      (1 to numRealInputs - 1).foreach { j =>
+        assert(Math.abs(inputs(i)(j).asInstanceOf[Double] - unrotatedInputs(i)(j).asInstanceOf[Double]) < 1e-9, "Unexpected modification of coordinate after applying transpose.")
+      }
+    }
+  }
+
+  /**
+    * Guess the mean should be invariant under rotation.
+    */
+  @Test
+  def testRotatedGTM(): Unit = {
+    val learner = GuessTheMeanLearner()
+    val model = learner.train(data).getModel()
+    val result = model.transform(data.map(_._1)).getExpected()
+
+    val rotatedLearner = FeatureRotator(GuessTheMeanLearner())
+    val rotatedModel = rotatedLearner.train(data).getModel()
+    val rotatedResult = rotatedModel.transform(data.map(_._1)).getExpected()
+
+    result.zip(rotatedResult).foreach { case (free: Double, rotated: Double) =>
+      assert(Math.abs(free - rotated) < 1.0e-9, s"${free} and ${rotated} should be the same")
+    }
+  }
+
+  /**
+    * Linear regression should be invariant under rotation
+    */
+  @Test
+  def testRotatedLinear(): Unit = {
+    val learner = LinearRegressionLearner()
+    val model = learner.train(data, Some(weights)).getModel()
+    val result = model.transform(data.map(_._1))
+    val expected = result.getExpected()
+    val gradient = result.getGradient()
+
+    val rotatedLearner = FeatureRotator(learner)
+    val rotatedModel = rotatedLearner.train(data, Some(weights)).getModel()
+    val rotatedResult = rotatedModel.transform(data.map(_._1))
+    val rotatedExpected = rotatedResult.getExpected()
+    val rotatedGradient = rotatedResult.getGradient()
+
+    expected.zip(rotatedExpected).foreach { case (free: Double, rotated: Double) =>
+      assert(Math.abs(free - rotated) < 1.0e-9, s"${free} and ${rotated} should be the same")
+    }
+
+    gradient.get.zip(rotatedGradient.get).foreach { case (free, rotated) =>
+      val diff = free.zip(rotated).map { case (f, s) => Math.abs(f - s) }.max
+      assert(diff < 1.0e-9, "Gradients should be the same")
+    }
+  }
+
+  /**
+    * Ridge regression should not depend on rotation
+    */
+  @Test
+  def testRotatedRidge(): Unit = {
+    val learner = LinearRegressionLearner(regParam = Some(1.0))
+    val model = learner.train(data).getModel()
+    val result = model.transform(data.map(_._1)).getExpected()
+
+    val rotatedLearner = new FeatureRotator(learner)
+    val rotatedModel = rotatedLearner.train(data).getModel()
+    val rotatedResult = rotatedModel.transform(data.map(_._1)).getExpected()
+
+    result.zip(rotatedResult).foreach { case (free: Double, rotated: Double) =>
+      assert(Math.abs(free - rotated) < 1.0e-9, s"${free} and ${rotated} should be the same")
+    }
+  }
+
+  /**
+   * Regression trees should not depend on rotation
```

Comment:
Regression trees _should_ depend on the rotation, but they should memorize their inputs regardless (if grown to full depth).  Could you add a check that non-training inputs do change?